libs/server/Storage/Session/ObjectStore/SortedSetGeoOps.cs

```diff
@@ -41,5 +44,160 @@ public GarnetStatus GeoCommands<TObjectContext>(byte[] key, ref ObjectInput inpu
+
+                var searchOutHandler = searchOutMem.Memory.Memory.Pin();
+                try
+                {
+
+                    var searchOutPtr = (byte*)searchOutHandler.Pointer;
+                    var currOutPtr = searchOutPtr;
+                    var endOutPtr = searchOutPtr + searchOutMem.Length;
+
+                    if (RespReadUtils.TryReadErrorAsSpan(out var error, ref currOutPtr, endOutPtr))
+                    {
+                        while (!RespWriteUtils.WriteError(error, ref curr, end))
+                            ObjectUtils.ReallocateOutput(ref output, ref isMemory, ref ptr, ref ptrHandle, ref curr, ref end);
+                        return GarnetStatus.OK;
+                    }
+
+                    var destinationKey = destination.ToArray();
+                    objectStoreLockableContext.Delete(ref destinationKey);
+
+                    RespReadUtils.ReadUnsignedArrayLength(out var foundItems, ref currOutPtr, endOutPtr);
+
+                    // Prepare the parse state for sorted set add
+                    var zParseState = new SessionParseState();
+                    zParseState.Initialize(foundItems * 2);
+
+                    for (int j = 0; j < foundItems; j++)
+                    {
+                        RespReadUtils.ReadUnsignedArrayLength(out var innerLength, ref currOutPtr, endOutPtr);
+                        Debug.Assert(innerLength == 2, "Should always has location and hash or distance");
+
+                        RespReadUtils.TrySliceWithLengthHeader(out var location, ref currOutPtr, endOutPtr);
+                        if (isStoreDist)
+                        {
+                            RespReadUtils.ReadSpanWithLengthHeader(out var score, ref currOutPtr, endOutPtr);
+                            zParseState.SetArgument(2 * j, ArgSlice.FromPinnedSpan(score));
+                            zParseState.SetArgument((2 * j) + 1, ArgSlice.FromPinnedSpan(location));
+                        }
+                        else
+                        {
+                            RespReadUtils.ReadIntegerAsSpan(out var score, ref currOutPtr, endOutPtr);
+                            zParseState.SetArgument(2 * j, ArgSlice.FromPinnedSpan(score));
+                            zParseState.SetArgument((2 * j) + 1, ArgSlice.FromPinnedSpan(location));
+                        }
+                    }
+
+                    // Prepare the input
+                    var zAddInput = new ObjectInput(new RespInputHeader
+                    {
+                        type = GarnetObjectType.SortedSet,
+                        SortedSetOp = SortedSetOperation.ZADD,
+                    }, ref zParseState);
+
+                    var zAddOutput = new GarnetObjectStoreOutput { spanByteAndMemory = new SpanByteAndMemory(null) };
+                    RMWObjectStoreOperationWithOutput(destinationKey, ref zAddInput, ref objectStoreLockableContext, ref zAddOutput);
```

Comment:
Why do another call to storage? Why not have 1 call that does both?

Reply:
What do you mean by "both"? Delete and Insert? I am not sure how I can do that because the key we need to delete can be of a different type (Anything other than SortedSet) than the one we need to insert (SortedSet)

Reply:
Ah yes it's a multi-key operation, you're right.